### DISCUSSION TITLE
What is the purpose of 'copyq.log.lock' file?

### DISCUSSION THREAD

**Discussion by Shakil-Shahadat** [hluk/CopyQ#1795] [Q&A] [answered]
On windows, it exists in the '%AppData%\copyq\copyq' folder.

As well as, what are the purposes of 'copyq.lock' and 'copyq-commands.ini.new.bak' files in the '%AppData%\copyq\' folder.

**Comment by hluk** [comment, marked as answer]
`copyq.lock` is a lock file to ensure at most single app session is running and to avoid multiple subprocesses to write to the log simultaneously. This mechanism is new in 5.0.0:  #1738

Files `*.bak` (or `*-bak.conf`) are created as back up for configuration. There were some problems in past with the configuration file being emptied. It was probably some bug in Qt framework and I'm not sure if this has been fixed already.

File `copyq-commands.ini.new.bak` was created during one-time migration of the commands to a separate file. The file not being deleted is a **bug**. Fix:  #1797

**Comment by Shakil-Shahadat** [comment]
Thank you so much for your hard work.
